### ISSUE TITLE
Ubuntu xenial kernel panic in bpf_map_update_elem using ext4slower

### ISSUE THREAD

**Issue by stefreak** [iovisor/bcc#1678]
We reported a bug to Launchpad: https://bugs.launchpad.net/ubuntu/+source/linux-meta-lts-xenial/+bug/1763352

Using the latest Xenial LTS kernel we get a kernel panic using bcc tools.

I am not sure if this is appropriate here, but maybe there are more people affected by this – then we'd be glad for people clicking on "This affects me" to bump the bug heat.

If you know a workaround or wether this bug exists in other distributions as well would also be highly appreciated.

**Comment by stefreak** [comment]
We found out that this is an Ubuntu specific issue and submitted a patch to Launchpad to fix the crash.

**Comment by stefreak** [comment]
Now marked as duplicate of https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1763454

Canonical is working on it 👍

**Comment by brent80** [comment]
"This affect me" but glad to know someone is working on it

**Comment by stefreak** [comment]
Fix is committed :) Will be in the next release. Once it landed I will close this issue.

**Comment by brent80** [comment]
I updated today to see if it was solved with this release but it didn't work for me.
4.4.0-122-generic

```
λ ~/bcc/examples/ master uname -r
4.4.0-122-generic
λ ~/bcc/examples/ master sudo ./hello_world.py
bpf: Failed to load program: Unknown error 524
0: (b7) r1 = 2593
1: (6b) *(u16 *)(r10 -4) = r1
2: (b7) r1 = 1684828783
3: (63) *(u32 *)(r10 -8) = r1
4: (18) r1 = 0x57202c6f6c6c6548
6: (7b) *(u64 *)(r10 -16) = r1
7: (b7) r1 = 0
8: (73) *(u8 *)(r10 -2) = r1
9: (bf) r1 = r10
10: (07) r1 += -16
11: (b7) r2 = 15
12: (85) call 6
13: (b7) r0 = 0
14: (95) exit

Traceback (most recent call last):
  File "./hello_world.py", line 11, in <module>
    BPF(text='int kprobe__sys_clone(void *ctx) { bpf_trace_printk("Hello, World!\\n"); return 0; }').trace_print()
  File "/usr/lib/python2.7/dist-packages/bcc/__init__.py", line 308, in __init__
    self._trace_autoload()
  File "/usr/lib/python2.7/dist-packages/bcc/__init__.py", line 916, in _trace_autoload
    fn = self.load_func(func_name, BPF.KPROBE)
  File "/usr/lib/python2.7/dist-packages/bcc/__init__.py", line 348, in load_func
    (func_name, errstr))
Exception: Failed to load BPF program kprobe__sys_clone: Unknown error 524```

**Comment by stefreak** [comment]
@brent80 interesting, did this work for you before `4.4.0-119`? Because if yes that's now a different regression, not a kernel panic.

**Comment by stefreak** [comment]
@brent80 I can't explain your error message but AFAIK the fixed version is not released yet (will be `4.4.0-123`). A colleague of mine tested a prerelease version and it fixed the issue for us apparently. You can get the prerelease version via the proposed track: https://wiki.ubuntu.com/Testing/EnableProposed

**Comment by brent80** [comment]
@stefreak Ok.  Yes it worked for me in 116, but not 119 or 122, but as you said I thought it would have been released now, but I will wait until 123

**Comment by brent80** [comment]
So this is still an issue for me, is this a different bug?  If I load kernel 116 it works just fine.

```
λ ~/ uname -r
4.4.0-124-generic
λ ~/ cd bccλ ~/bcc/ master ls
cmake           CODEOWNERS               COPYRIGHT.txt  Dockerfile.debian  docs      FAQ.txt  INSTALL.md     LICENSE.txt  man            README.md  snapcraft  src    tools
CMakeLists.txt  CONTRIBUTING-SCRIPTS.md  debian         Dockerfile.ubuntu  examples  images   introspection  LINKS.md     QUICKSTART.md  scripts    SPECS      tests
λ ~/bcc/ master cd examples 
λ ~/bcc/examples/ master ls
CMakeLists.txt  cpp  hello_world.py  lua  networking  tracing  usdt_sample
λ ~/bcc/examples/ master sudo ./hello_world.py 
bpf: Failed to load program: Unknown error 524
0: (b7) r1 = 2593
1: (6b) *(u16 *)(r10 -4) = r1
2: (b7) r1 = 1684828783
3: (63) *(u32 *)(r10 -8) = r1
4: (18) r1 = 0x57202c6f6c6c6548
6: (7b) *(u64 *)(r10 -16) = r1
7: (b7) r1 = 0
8: (73) *(u8 *)(r10 -2) = r1
9: (bf) r1 = r10
10: (07) r1 += -16
11: (b7) r2 = 15
12: (85) call 6
13: (b7) r0 = 0
14: (95) exit

Traceback (most recent call last):
  File "./hello_world.py", line 12, in <module>
    BPF(text='int kprobe__sys_clone(void *ctx) { bpf_trace_printk("Hello, World!\\n"); return 0; }').trace_print()
  File "/usr/lib/python2.7/dist-packages/bcc/__init__.py", line 308, in __init__
    self._trace_autoload()
  File "/usr/lib/python2.7/dist-packages/bcc/__init__.py", line 916, in _trace_autoload
    fn = self.load_func(func_name, BPF.KPROBE)
  File "/usr/lib/python2.7/dist-packages/bcc/__init__.py", line 348, in load_func
    (func_name, errstr))
Exception: Failed to load BPF program kprobe__sys_clone: Unknown error 524```

**Comment by damienrg** [comment]
@brent80 I had the same problem and it is fixed in 4.4.0.125-generic available in proposed.

**Comment by stefreak** [comment]
Launchpad says it is fixed now in 4.4.0-127.153 – did anyone have a chance already to test that version?

**Comment by damienrg** [comment]
Yes, it is fixed for me with 4.4.0-127.153.

**Comment by stefreak** [comment]
Cool then i'll close this issue
